plugins/version-control-systems/git/src/main/kotlin/Git.kt

```diff
@@ -82,6 +82,8 @@ object GitCommand : CommandLineTool {
         }.orEmpty()
 }
 
+private val SHA1_REGEX = Regex("^[0-9a-f]{40}$")
```

Comment:
In other places we use `[0-9a-fA-F]{7,40}`, which is a bit safer as it accepts uppercase chars, and at a minimum 7 chars.

Reply:
I tried with 7 chars, but this throws an exception because it must be an exact match.If it is not (like with only 7 chars of the SHA1), the fallback to fetch all revisions was invoked anyway and then afterwards the checkout with the short SHA1 worked.
Regarding uppercase chars, I haven't tried, but I would assume it is the same, except if Git itself is ignoring the case. I just wonder if this is relevant because where would such a SHA1 with uppercase characters come from?

Reply:
> it must be an exact match.

Ah, right, I now [remember that requirement](https://stackoverflow.com/questions/31278902/how-to-shallow-clone-a-specific-commit-with-depth-1/43136160#comment126531094_43136160).

> except if Git itself is ignoring the case.

It is, I've just checked.

> I just wonder if this is relevant because where would such a SHA1 with uppercase characters come from?

From user input on the CLI or a server UI form?

Reply:
> From user input on the CLI or a server UI form?

What I meant is that people are not writing SHA1s by hand but copy pasting them from somewhere, and I don't know any tool that would show Git SHA1s with uppercase characters. But it's no problem changing the regex anyway.